lib/modules/manager/tekton/types.ts

```diff
@@ -1,15 +1,19 @@
 export interface TektonResource {
   spec: TektonResourceSpec;
-  items: TektonResource[];
+  items?: TektonResource[];
 }
 
 interface TektonResourceSpec {
   // TaskRun
   taskRef: TektonBundle;
   // PipelineRun
   pipelineRef: TektonBundle;
+  // PipelienRun with inline Pipeline definition
+  pipelineSpec?: TektonResourceSpec;
   // Pipeline
   tasks: TektonResourceSpec[];
+  // Pipeline
+  finally: TektonResourceSpec[];
```

Comment:
```suggestion
  finally?: TektonResourceSpec[];
```
isn't that optional?